bsp/Src/CAN.c

```diff
@@ -178,10 +178,11 @@ static const uint32_t can3_recv_entry_count = 0;
 #endif /* can3_recv_entries.h */
 #endif /* CAN3 */
 
-static inline void HAL_CAN_MspF4Init(CAN_HandleTypeDef* hcan){
+// CAN GPIO init, can be overridden by user
+__weak void HAL_CAN_MspGPIOInit(CAN_HandleTypeDef* hcan) {
```

Comment:
The weak function def works, but there are some edge cases I can think of (which are mainly just user error) that could break this. For example, user defines HAL_CAN_MspGPIOInit but not GPIODeInit and then deinit is called by smth in like a fault state. Maybe that wouldn't be the case or maybe there's something else, but since it's user error you can probably just have the user fix it?

Anyways you can ignore this comment I just wanted this to be noted. Perhaps if this sort of edge case ever presents a problem a function pointer could be used instead of a weak user def; where the user passes in a function pointer to the init function for gpio init and it's just called in the MspGPIOinit def that's made here. but not necessary now (or hopefully never!)